.github/changelog-configuration.json

```diff
@@ -0,0 +1,20 @@
+{
+    "categories": [
+        {
+            "title": "## 🚀 Features",
+            "labels": ["enhancement", "feature"]
+        },
+        {
+            "title": "## ⚠️ (Breaking) Changes",
+            "labels": ["change"]
```

Comment:
Should we make a further separation between non-breaking changes and breaking changes?
If so, what should the labels be named? Can we even keep this discipline? Ideally, only major releases should get breaking changes (hence the major release), but this would also list general, probably unrelated/uninteresting changes like "Refactor backup logic".

Or should we remove "(Breaking)" from the Title altogether and not make a distinction, relying on any official migration guides in the documentation?

Reply:
I'd say we stick to the semver as much as possible, so only major releases should break stuff. Which would make the breaking changes here obsolete.

Reply:
Going by semver alone would also mean that we don't categorize documentation PRs (even simple typos etc). I personally like a distinction between "changes that are infrastructure/quality (git, pipelines, makefile etc)" related and actual, new features of K8up itself. Now we could argue to ignore all quality or infra related stuff that aren't strictly related to semver, but I'd prefer to have them listed too, but not under bugfixes or new features.

Reply:
Also, there's a lot of merged Renovate PRs. I think for those alone it's worth to have an "uncategorized" or "dependency update" section. I doubt that we'll be able to label all Renovate PRs, as they're merged pretty fast while forgetting to set a label.

Reply:
I'd remove the "(Breaking)" from the title and find another, less intrusive icon. Breaking changes should be documented separately if there are any, in a different category (label). I would also only mark changes "breaking" if there is any user-interaction required when upgrading K8up.

Listing infrastructure updates in a separate category would be cool, as well as documentation and dependency updates.

> I doubt that we'll be able to label all Renovate PRs
Afaik Renovate can be configured to automatically set labels.

Categorizing things in the changelog allows the reader to distinguish the changes and decide if they are interesting to the reader or not and can skip it then.

For inspiration: Other projects using icons in releases:
* https://github.com/TryGhost/Ghost/releases
* https://github.com/RocketChat/Rocket.Chat/releases/

Reply:
Right, I didn't think about those infrastructure fixes.

Listing them separately would be cool, I like how rocketchat does their changelog!

Reply:
> Afaik Renovate can be configured to automatically set labels.

Neat, didn't know! Adjusted PR with category.

I have categorized Infrastructure-related changes as "minor changes" and made a separate for breaking changes. This time with a magnifying glass, indicating that this is a section users should look into. Any other proposals?

Reply:
Looks good, I'd say go for it :+1:

Reply:
I'm thinking about better title for breaking changes. Coming up with a "weaker" word for "Breaking" is not that easy:
* "Notable changes" (goes well with magnifying glass icon, but maybe too weak)
* "user-facing changes" (not my favorite)
* "Changes (Action needed)" or "Changes (Migration needed)"

any other? or we keep "breaking changes"?

Reply:
Well for breaking changes I'd keep the breaking to be honest.

Reply:
Ok, I think this is good enough for now :)